scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -579,18 +579,21 @@
    given a job, its resources, its task-id and a function assigned-cotask-getter. assigned-cotask-getter should be a
    function that takes a group uuid and returns a set of task-ids, which correspond to the tasks that will be assigned
    during the same Fenzo scheduling cycle as the newly created TaskRequest."
-  [db job & {:keys [resources task-id assigned-resources guuid->considerable-cotask-ids running-cotask-cache]
+  [db job & {:keys [resources task-id assigned-resources guuid->considerable-cotask-ids running-cotask-cache job-uuid->reserved-host]
           :or {resources (util/job-ent->resources job)
                task-id (str (java.util.UUID/randomUUID))
                assigned-resources (atom nil)
                guuid->considerable-cotask-ids (constantly #{})
-               running-cotask-cache (atom (cache/fifo-cache-factory {} :threshold 1))}}]
-  (let [constraints (into (constraints/make-fenzo-job-constraints job)
+               running-cotask-cache (atom (cache/fifo-cache-factory {} :threshold 1))
+               job-uuid->reserved-host {}}}]
+  (let [constraints (-> (constraints/make-fenzo-job-constraints job)
+                        (into
                           (remove nil?
                                   (mapv (fn make-group-constraints [group]
                                           (constraints/make-fenzo-group-constraint
                                            db group #(guuid->considerable-cotask-ids (:group/uuid group)) running-cotask-cache))
                                         (:group/_job job))))
+                        (conj (constraints/build-rebalancer-reservation-constraint job job-uuid->reserved-host)))
```

Comment:
Should we make sure that cheaper to compute constraints are placed earlier in the list of constraints? Would it be helpful if the `rebalancer-reservation-constraint` was conj-ed earlier before the `into`?

Reply:
Done